lib/rouge/lexers/gherkin.rb

```diff
@@ -19,8 +19,7 @@ def self.detect?(text)
 
       # self-modifying method that loads the keywords file
       def self.keywords
-        load File.join(__dir__, 'gherkin/keywords.rb')
-        keywords
+        @keywords ||= instance_eval(File.read(File.join(__dir__, "gherkin/keywords.rb")))
```

Comment:
@pyrmont This is change in behavior.
By using `load` previously, the contents of `'gherkin/keywords.rb'` would be re-evaluated on every call (technically, allowing `rouge.jneen.net` and the visual test app pick up changes during development which I believe is the primary reason @jneen refrained from using `require` and went with `load`.

Memoizing the method breaks that contract.

Reply:
@jneen would have to speak about that. The files read from here are currently generated from a Rake file and memoise their contents. Without background knowledge, I'd say that hot reloading of these files doesn't seem to be how they're intended to be used.

For what it's worth, I wouldn't characterise these classes as having a 'contract' to break. These are internal implementation details as far as I can see and don't provide a public-facing API to consumers.

Reply:
It does not break that contract, as the visual test app drops the `Rouge` constant from the global namespace. This is a really good change.

Reply:
The comment should be changed, though. I don't think you could call this a 'self-modifying method' any more. That's probably true for all the other similar methods I edited.

Reply:
(also @ashmaroli given that it was "self-modifying", loading the file once changed the definition of the method, so... it wouldn't get reevaluated)

Reply:
Though changing a method definition at runtime is a perf no-no anyways, since it drops the entire method lookup cache for the whole runtime, so this is a much better approach anyways

Reply:
Okay. Thanks for the clarifications.